charts/jaeger/templates/allInOne-config.yaml

```diff
@@ -0,0 +1,12 @@
+{{ if .Values.userconfig }}
```

Comment:
I think this file should be called `user-config.yaml`, it's not specific to all-in-one.

Reply:
i used that only in the start but I got this error 
`Error: INSTALLATION FAILED: parse error at (jaeger/templates/allInOne-config.yaml:1): bad character U+002D '-'`
the problem is helm chart don't allow '-' there  is workaround we can use 
`{{index .Values.user-config }}`
but with if or any other condition statements it throw this error 
`Error: INSTALLATION FAILED: parse error at (jaeger/templates/allInOne-config.yaml:1): unexpected <if> in operand`

Reply:
I made this change to the template
```diff
diff --git a/charts/jaeger/templates/allInOne-config.yaml b/charts/jaeger/templates/allInOne-config.yaml
index eca21f2..e473618 100644
--- a/charts/jaeger/templates/allInOne-config.yaml
+++ b/charts/jaeger/templates/allInOne-config.yaml
@@ -8,5 +8,5 @@ metadata:
     {{- include "jaeger.labels" . | nindent 4 }}
 data:
   config.yaml: |
-        {{ .Values.userconfig | nindent 8 }}
+        {{- .Values.userconfig | nindent 4 }}
 {{- end }}
```

created mock config file
```
$ cat myconfig.yaml
hello:
  world: true
```
and ran a test 
```
$ helm template --set-file userconfig=myconfig.yaml charts/jaeger > test-output.txt
```

which resulted in reasonably looking resource file:
```yaml
# Source: jaeger/templates/allInOne-config.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: jaeger-configmap
  namespace: default
  labels:
    helm.sh/chart: jaeger-4.0.0
    app.kubernetes.io/name: jaeger
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "2.0.0-rc2"
    app.kubernetes.io/managed-by: Helm
data:
  config.yaml: |
    hello:
      world: true
```